test/test-bopomofo.c

```diff
@@ -1690,6 +1691,63 @@ void test_KB()
     test_KB_MPS2();
 }
 
+void test_chewing_phone_to_bopomofo()
+{
+    char *u8phone;
+    char *rt;
+    int  len;
+    uint16_t phone;
+    uint16_t expect;
+    /*
+    the libchewing divides a completed bopomofo into 4 parts,
+
+        1st part: ㄅㄆㄇㄈㄉㄊㄋㄌㄍㄎㄏㄐㄑㄒㄓㄔㄕㄖㄗㄘㄙ
+        2nd part: ㄧㄨㄩ
+        3rd part: ㄚㄛㄜㄝㄞㄟㄠㄡㄢㄣㄤㄥㄦ
+        4th part:  ˙ˊˇˋ
+
+    calculates each part's offset and stores into a 16-bit unsigned by following rule:
+    
+    16-bit unsinged = ( 1st part offset )<<9 + ( 2nd part offset )<<7 + ( 3rd part offset )<<3 + (4th part offset),
+    e.g., ㄆㄣ , 1st part offset = 2, 2nd part offset = 0, 3rd part offset = 10, 4th part offset = 0, 
+    the number for  ㄆㄣ is (2<<9)+(0<<7)+(10<<3)+(0) = 1104 
+    */
+
+    start_testcase(NULL, fd);
+
+    u8phone = "\xE3\x84\x86\xE3\x84\xA3" /* ㄆㄣ */ ;
+    phone = UintFromPhone(u8phone);
+    expect = (2<<9)+(0<<7)+(10<<3)+(0);
```

Comment:
I meant, we prefer `expect = (2 << 9) + (0 << 7) + (10 << 3) + (0);`. We use spaces to split the items.